fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -2480,6 +1690,127 @@ private static void checkPositive(int value, String desc) {
         }
     }
 
+    /**
+     * A builder for the  indexFromIndex policy. Let the caller set a source index and a fallback policy.
+     */
+    public static class IndexFromIndexPolicy {
+        public static final IndexFromIndexPolicy INACTIVE = new IndexFromIndexPolicy();
+        @Nullable private final String sourceIndex;
+        private final boolean allowRecordScan;
+        private boolean fallback = false;
```

Comment:
It would be better to keep the user's configuration (an immutable object containing `sourceIndex` and `allowRecordScan`) and mutable internal state (`fallback`) separate.

Reply:
I'm a little confused - since the user must use the IndexFromIndexPolicy.Builder (by your suggestion), this is not exposed to him.

Reply:
I'm not sure what the specific ask was before, but I think separating the configuration (i.e., the policy) from the state (i.e., whether `fallback` was used or not) would make sense. I don't really think the user should have to provide a `Builder`, though, and indeed, it looks like `.setIndexFromIndex` takes a non-`Builder`